tlv_make_data: undefined behavior

https://travis-ci.org/yoursunny/esp8266ndn/jobs/579081564
```
In file included from /home/travis/Arduino/libraries/esp8266ndn/src/ndn-lite/encode/wrapper-api.c:16:0:
/home/travis/Arduino/libraries/esp8266ndn/src/ndn-lite/encode/wrapper-api.c: In function 'tlv_make_data':
/home/travis/Arduino/libraries/esp8266ndn/src/ndn-lite/encode/wrapper-api.c:85:26: warning: 'enum TLV_DATAARG_TYPE' is promoted to 'int' when passed through '...'
     argtype = va_arg(vl, enum TLV_DATAARG_TYPE);
                          ^
/home/travis/Arduino/libraries/esp8266ndn/src/ndn-lite/encode/wrapper-api.c:85:26: note: (so you should pass 'int' not 'enum TLV_DATAARG_TYPE' to 'va_arg')
/home/travis/Arduino/libraries/esp8266ndn/src/ndn-lite/encode/wrapper-api.c:85:26: note: if this code is reached, the program will abort
```

See https://stackoverflow.com/a/23983508 for an explanation on why this is undefined behavior, and how to fix it.

--- a/encode/wrapper-api.c
+++ b/encode/wrapper-api.c
@@ -82,7 +82,7 @@ tlv_make_data(uint8_t* buf, size_t buflen, size_t* result_size, int argc, ...)
 
   va_start(vl, argc);
   for (i = 0; i < argc && ret == NDN_SUCCESS; i++) {
-    argtype = va_arg(vl, enum TLV_DATAARG_TYPE);
+    argtype = va_arg(vl, int);
     switch(argtype) {
       case TLV_DATAARG_NAME_PTR:
         arg_ptr = va_arg(vl, void*);
@@ -265,7 +265,7 @@ tlv_parse_data(uint8_t* buf, size_t buflen, int argc, ...)
   // Parse args
   va_start(vl, argc);
   for(i = 0; i < argc && ret == NDN_SUCCESS; i ++) {
-    argtype = va_arg(vl, enum TLV_DATAARG_TYPE);
+    argtype = va_arg(vl, int);
     switch(argtype) {
       case TLV_DATAARG_NAME_PTR:
         arg_ptr = va_arg(vl, ndn_name_t*);
@@ -511,7 +511,7 @@ tlv_make_interest(uint8_t* buf, size_t buflen, size_t* result_size, int argc, ..
   ndn_interest_init(&interest);
   va_start(vl, argc);
   for(i = 0; i < argc && ret == NDN_SUCCESS; i ++) {
-    argtype = va_arg(vl, enum TLV_INTARG_TYPE);
+    argtype = va_arg(vl, int);
     switch(argtype) {
       case TLV_INTARG_NAME_PTR:
         arg_ptr = va_arg(vl, void*);
@@ -679,7 +679,7 @@ tlv_parse_interest(uint8_t* buf, size_t buflen, int argc, ...)
   // Parse args
   va_start(vl, argc);
   for (i = 0; i < argc && ret == NDN_SUCCESS; i++) {
-    argtype = va_arg(vl, enum TLV_INTARG_TYPE);
+    argtype = va_arg(vl, int);
     switch(argtype) {
       case TLV_INTARG_NAME_PTR:
         arg_ptr = va_arg(vl, ndn_name_t*);

